graphx/src/main/scala/org/apache/spark/graphx/util/PeriodicGraphCheckpointer.scala

```diff
@@ -87,10 +88,7 @@ private[mllib] class PeriodicGraphCheckpointer[VD, ED](
 
   override protected def persist(data: Graph[VD, ED]): Unit = {
     if (data.vertices.getStorageLevel == StorageLevel.NONE) {
-      data.vertices.persist()
-    }
-    if (data.edges.getStorageLevel == StorageLevel.NONE) {
-      data.edges.persist()
+      data.persist()
```

Comment:
hmm, `GraphImpl.persist` actually persists its `vertices` and `replicatedVertexView.edges` (i.e., `edges`). But the problem is we only check the storage level of vertices. Maybe we should keep the original.